### PR TITLE
[BACKLOG-21225] Add authentication to MQTT Consumer

### PR DESCRIPTION
**Pull request by DFieldFL**

### CODE REVIEW

**Comment by DFieldFL** [comment]
@pentaho/ackbar please review

File: kettle-plugins/mqtt/src/main/java/org/pentaho/di/trans/step/mqtt/MQTTStreamSource.java [hunk 1]
```diff
@@ -79,9 +76,12 @@
         .withQos( mqttConsumerMeta.getQos() )
         .withStep( mqttConsumer )
         .withCallback( callback )
+        .withUsername( mqttConsumerMeta.getUsername() )
```
Comment (kurtwalker) [inline kettle-plugins/mqtt/src/main/java/org/pentaho/di/trans/step/mqtt/MQTTStreamSource.java]: Should these be environment substituted?

Comment (DFieldFL) [reply kettle-plugins/mqtt/src/main/java/org/pentaho/di/trans/step/mqtt/MQTTStreamSource.java]: yes!

File: kettle-plugins/mqtt/src/main/java/org/pentaho/di/trans/step/mqtt/MQTTClientBuilder.java [hunk 2]
```diff
@@ -178,11 +179,11 @@ private MqttConnectOptions getOptions() {
     if ( isSecure ) {
       setSSLProps( options );
     }
-    if ( username != null ) {
-      options.setUserName( username );
+    if ( !StringUtil.isEmpty( username ) ) {
+      options.setUserName( step.environmentSubstitute( username ) );
```
Comment (DFieldFL) [inline kettle-plugins/mqtt/src/main/java/org/pentaho/di/trans/step/mqtt/MQTTClientBuilder.java]: @kurtwalker variable substitution fixed in the builder.

**Comment by DFieldFL** [comment]
@pentaho/ackbar please review

**Pull request merged**